transmart-rest-api-e2e/src/test/groovy/tests/rest/access/AccessCheckSpec.groovy

```diff
@@ -0,0 +1,103 @@
+package tests.rest.access
+
+import base.RESTSpec
+import config.OauthAdapter
+import groovy.json.JsonSlurper
+import static base.ContentTypeFor.JSON
+import static config.Config.*
+
+class AccessCheckSpec extends RESTSpec {
+    /**
+     * when: "I try to fetch all studies without access token"
+     * then: "I am not authorized to access the resource"
+     *
+     * when: "I try to fetch all studies with invalid access token"
+     * then: "I am not authorized to access the resource"
+     */
+    def "access to resources is blocked with an invalid token"() {
+        when: "I try to fetch all studies without access token"
+        def responseData = get([
+                path      : PATH_STUDIES,
+                acceptType: JSON,
+                user      : UNRESTRICTED_USER,
+                skipOauth : true,
+                statusCode: 401
+        ])
+        then: "I am not authorized to access the resource"
+        responseData.message == "Unauthorized"
+
+        when: "I try to fetch all studies with invalid access token"
+        def responseDataInvalidToken = responseData
+        responseDataInvalidToken.token = 'invalidToken'.bytes.encodeBase64().toString()
+        then: "I am not authorized to access the resource"
+        responseDataInvalidToken.message == "Unauthorized"
+
+    }
+
+    /**
+     * when: "I try to fetch all studies with a proper access token"
+     * then: "I have an access to the resource"
+     *
+     * when: "I fabricate the token by replacing original user data with a new user"
+     * then: "The signature is verified as invalid, I am not authorized to access the resource"
+     */
+    def "access token validation restricted per user"() {
+
+        when: "I try to fetch all studies with a proper access token"
+        def tokenForUnrestrictedUser = OauthAdapter.getToken(UNRESTRICTED_USER)
+
+        then: "I have an access to the resource"
+        get([
+                path      : PATH_STUDIES,
+                acceptType: JSON,
+                user      : UNRESTRICTED_USER,
+                skipOauth : true,
+                statusCode: 200,
+                token     : tokenForUnrestrictedUser
+        ])
+
+
+        when: "I fabricate the token by replacing original user data with a new user"
+        def tokenWithReplacedUser = replaceUserInToken(tokenForUnrestrictedUser, DEFAULT_USER)
+        def responseForNewUser = get([
+                path      : PATH_STUDIES,
+                acceptType: JSON,
+                user      : UNRESTRICTED_USER,
```

Comment:
Why do we have both `token` and `user`? `token` seems to be sufficient.
I don't fully understand where passing `UNRESTRICTED_USER` as the user and `DEFAULT_USER` in the token fails.

Reply:
you are right, user is not needed in this case, I'll remove it from this map.  It fails on checking the signature on the keycloak side. It compares the signature from the last part of the token with the `payload` part of the token, where there is a user data.